### PR TITLE
Fixes #30605 - replace mount_react_component in search

### PR DESCRIPTION
**Pull request by MariaAga**
replace mount_react_component with react_component in search

### CODE REVIEW

File: app/views/common/_searchbar.html.erb [hunk 1]
```diff
@@ -1,2 +1,3 @@
-<div id="search-bar"></div>
-<%= mount_search_bar("search-bar") %>
+<div id="search-bar">
```
Comment (amirfefer) [inline app/views/common/_searchbar.html.erb]: same thought from here https://github.com/theforeman/foreman/pull/7921

Comment (MariaAga) [reply app/views/common/_searchbar.html.erb]: :+1:

**Review by amirfefer — APPROVED** [review]
Thanks @MariaAga! :+1: 
Tested and works well

**Comment by tbrisker** [comment]
Looks like this change actually fixed a bug when navigating from non-react page to react page where the autocomplete url wasn't updating properly. :medal_sports: @MariaAga !

**Comment by tbrisker** [comment]
2.2 - 19e7a3004

**Pull request merged**